Mathlib/Tactic/Positivity/Basic.lean

```diff
@@ -73,6 +73,45 @@ such that `positivity` successfully recognises both `a` and `b`. -/
     pure (.nonzero (q(min_ne $pa' $pb') : Expr))
   | _, _ => pure .none
 
+/-- Extension for the `max` operator. The `max` of two numbers is nonnegative if at least one
+is nonnegative, strictly positive if at least one is positive, and nonzero if both are nonzero. -/
+@[positivity max _ _] def evalMax : PositivityExt where eval {u α} zα pα e := do
+  let .app (.app f (a : Q($α))) (b : Q($α)) ← withReducible (whnf e) | throwError "not max"
+  let result ← catchNone do
+    let _a ← synthInstanceQ (q(LinearOrder $α) : Q(Type u))
+    guard <|← withDefault <| withNewMCtxDepth <| isDefEq f q(max (α := $α))
```

Comment:
I tried moving these lines to the outside of the `catchNone` (to avoid duplication), but then I got the error:
```
application type mismatch
  orElse result
argument
  result
has type
  Strictness zα
    (Expr.app (Expr.app (Expr.const `SemilatticeInf.toPartialOrder [u]) α)
      (Expr.app (Expr.app (Expr.const `Lattice.toSemilatticeInf [u]) α)
        (Expr.app (Expr.app (Expr.const `DistribLattice.toLattice [u]) α)
          (Expr.app (Expr.app (Expr.const `instDistribLattice [u]) α) _a))))
    (Expr.app
      (Expr.app
        (Expr.app (Expr.app (Expr.const `Max.max [u]) α) (Expr.app (Expr.app (Expr.const `LinearOrder.toMax [u]) α) _a))
        a)
      b) : Type
but is expected to have type
  Strictness zα pα e : Type

```